src/mlpack/core/metrics/lmetric_impl.hpp

```diff
@@ -61,7 +61,19 @@ typename VecTypeA::elem_type LMetric<2, true>::Evaluate(
     const VecTypeA& a,
     const VecTypeB& b)
 {
-  return arma::norm(a - b, 2);
+  if ((std::is_same<typename VecTypeA::elem_type, double>::value or
```

Comment:
Do you think you can also add a test for this function when, e.g., `arma::Col<size_t>` or `arma::ucol` is used as the vector type?  You could just add it to the bottom of `metric_test.cpp`.  (As Conrad pointed out in another comment, this will fail to compile at the moment for the reasons detailed there.)

Reply:
Yeah definitely,  a couple of tests are required.